src/array/array.jl

```diff
@@ -467,6 +489,8 @@ function DimArray(f::Function, dim::Dimension; name=Symbol(nameof(f), "(", name(
      DimArray(f.(val(dim)), (dim,); name)
 end
 
+DimArray(itr::Base.Generator; kwargs...) = rebuild(collect(itr); kwargs...)
```

Comment:
Okay, tomorrow I'll test and work out any kinks tomorrow.

Reply:
The suggested change triggers:

```
ERROR: MethodError: similar(::Type{Matrix{Int64}}, ::Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}) is ambiguous.
```

when the following generator example is run

```
(x + y for x in 1:5, y in 1:3)
```

which is blocking development, and the previously working generator syntax method breaks with the same above error.

Do you know what's going on?

Reply:
Ok we're doing some subtle type piracy in `similar` because of the unions. That's unfortunate. We need to rewrite those methods so there is always a Dimensions not in a union in at least one position, so we own the methods and there is no ambiguity. 

Let's make a new issue for that (just your comment text here basically) and fix it later.

Reply:
I'm happy to open that new issue if you want.

Reply:
Thanks